src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5602,6 +5602,50 @@ public override IOperation VisitIsPattern(IIsPatternOperation operation, int? ca
                 operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
+        public override IOperation VisitInvalid(IInvalidOperation operation, int? captureIdForResult)
+        {
+            if (_currentStatement == operation)
+            {
+                return VisitInvalidOperationStatement(operation);
+            }
+            return VisitInvalidOperationExpression(operation);
+        }
+
+        private IOperation VisitInvalidOperationStatement(IInvalidOperation operation)
+        {
+            Debug.Assert(_currentStatement == operation);
+            VisitStatements(operation.Children);
+            return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
```

Comment:
> isn't our standard pattern for this to add this as a statement, and return null, rather than returning as we're visiting in a known statement context?

I do not think this is a violation, returning works too, by design. We use that for some simple cases like expression statement, etc.